test/unittest/unittest_nntrainer_layers.cpp

```diff
@@ -102,31 +102,29 @@ class nntrainer_abstractLayer : public ::testing::Test {
 
   void setBatch(unsigned int batch) { layer.setBatch(batch); }
 
+  void matchOutput(const float *result, const float *golden, size_t length) {
+    for (size_t i = 0; i < length; ++i) {
+      EXPECT_NEAR(result[i], golden[i], local_tolerance);
+    }
+  }
+
   void matchOutput(const nntrainer::Tensor &result,
                    const nntrainer::Tensor &golden) {
-    const float *out_ptr, *golden_ptr;
-
-    out_ptr = result.getData();
-    golden_ptr = golden.getData();
+    matchOutput(result.getData(), golden.getData(), result.length());
+  }
 
-    for (size_t i = 0; i < result.length(); ++i) {
-      EXPECT_NEAR(out_ptr[i], golden_ptr[i], local_tolerance);
-    }
+  void matchOutput(const float *result, const char *expected) {
+    nntrainer::Tensor golden;
+    loadFile(expected, golden);
+    matchOutput(result, golden.getData(), golden.length());
```

Comment:
FYI) golden.length() is 0. (It is a known issue left untouched before this patch as well)

Reply:
How about putting a comment here?

Reply:
Even after loading data in it?

Reply:
Yeah, because golden is uninitialized in this function, which has length of 0

Reply:
Some unittest might be wrong after fixing this. I didn't fix it on purpose

Reply:
I will add a comment here.